src/main/java/core/basesyntax/service/FruitTransaction.java

```diff
@@ -0,0 +1,57 @@
+package core.basesyntax.service;
+
+public class FruitTransaction {
+    private Operation operation;
+    private String fruit;
+    private int amount;
+
+    public Operation getOperation() {
+        return operation;
+    }
+
+    public void setOperation(Operation operation) {
+        this.operation = operation;
+    }
+
+    public String getFruit() {
+        return fruit;
+    }
+
+    public void setFruit(String fruit) {
+        this.fruit = fruit;
+    }
+
+    public int getAmount() {
+        return amount;
+    }
+
+    public void setAmount(int amount) {
+        this.amount = amount;
+    }
+
+    public enum Operation {
+        BALANCE("b"),
+        SUPPLY("s"),
+        PURCHASE("p"),
+        RETURN("r");
+
+        private String code;
+
+        Operation(String code) {
+            this.code = code;
+        }
+
+        public String getCode() {
+            return code;
+        }
+
+        public static Operation operationFromCode(String code) {
+            for (Operation operation : Operation.values()) {
+                if (operation.getCode().equals(code)) {
+                    return operation;
+                }
+            }
+            throw new RuntimeException("can't find operation");
+        }
+    }
```

Comment:
According to the SOLID principles, each class should have a single responsibility. Here, the `FruitTransaction` class is doing two things: it represents a fruit transaction and it also converts a string to an `Operation` enum. It would be better to move the `Operation` enum and the `operationFromCode` method to a separate class. This way, the `FruitTransaction` class will only be responsible for representing a fruit transaction, and the new class will be responsible for converting a string to an `Operation` enum.